blog/2021/02/julia-1.6-highlights.md

```diff
@@ -0,0 +1,276 @@
+goods, some releases just come out better than others and you can't always tell
+until after they are baked and eaten. The full list of changes can be found in
+the [NEWS file](https://github.com/JuliaLang/julia/blob/release-1.6/NEWS.md),
+but here we'll give a more in depth overview of some of the release highlights.
+
+\toc
+
+## Parallel precompilation
+
+_Ian Butterworth_
+
+Executing all of the statements in a module often involves compiling a large
+amount of code, so Julia creates precompiled caches of the module to reduce this
+time. In 1.6, this package precompilation is faster and happens before you leave
+`pkg>` mode. Until now, precompilation took place solely as a single-processed
+sequence, precompiling dependencies one-by-one when needed during the linear
+code loading process when a package is `using`/`import`-ed for the first time.
+
+The olden days of <= 1.5:
+```julia-repl
+(v1.5) pkg> add DifferentialEquations
+...
+julia> @time using DifferentialEquations
+[ Info: Precompiling DifferentialEquations [0c46a032-eb83-5123-abaf-570d42b7fbaa]
+  474.288251 seconds …
+```
+In 1.6, `pkg>` mode gains a heavily parallelized precompile operation that is
+auto-invoked after package actions, to keep the active environment ready to
+load.
+```julia-repl
+(v1.6) pkg> add DifferentialEquations
+...
+Precompiling project...
+  Progress [========================================>]  112/112
+112 dependencies successfully precompiled in 72 seconds
+
+julia> @time using DifferentialEquations
+  4.995477 seconds …
+```
+Whereas the previous code loading precompilation process took ~8 minutes to
+precompile and load without indicating progress while doing so, the new
+mechanism took 1m12s to precompile while showing progress through the
+dependencies, then the first time the package is loaded it happens at full
+speed. The new parallel precompilation process takes a depth-first approach of
+working through the dependency tree in the manifest, first precompiling the
+packages with no dependencies and working upward to the packages listed in the
+environment’s `Project.toml`, allowing multiple packages to be precompiled at
+the same time. The operation is multi-processed, as opposed to multi-threaded,
+so is not limited by Julia’s thread count. By default, Julia will spawn a
+maximum CPU-balanced load of package precompile jobs at once based on the number
+of CPU cores. Errors during precompilation will only throw for packages listed
+in the Project to allow for dependencies that may be listed in manifests but not
+loaded, and the auto precompilation process will remember if a package has
+errored within the given environment and will not retry until it changes.
+Auto-precompilation can be gracefully interrupted with a `ctrl-c` and disabled
+by setting the environment variable `JULIA_PKG_PRECOMPILE_AUTO=0`.
+
+~~~
+<script id="asciicast-381203" src="https://asciinema.org/a/381203.js" async></script>
```

Comment:
Fit on page better & smaller (I don't have commit rights, so would someone mind accepting this)
```suggestion
<script id="asciicast-381203" src="https://asciinema.org/a/381203.js" data-rows="25" data-cols="104" async></script>
```